src/storage/data_table.cpp

```diff
@@ -180,26 +223,40 @@ bool DataTable::ScanBaseTable(Transaction &transaction, DataChunk &result, Table
 		current_row += STANDARD_VECTOR_SIZE;
 		return true;
 	}
-
+	SelectionVector sel;
```

Comment:
I would still like a special case here where if there are no filters and there are no deleted entries you use the old way of scanning (i.e. calling Sequence for row ids, Scan for regular columns).